.github/workflows/gradle.yml

```diff
@@ -29,18 +29,24 @@ on:
 # Define a few constants that are shared across all the jobs.
 env:
   GLOBAL_CACHE_PATH: |
-    .gradle
     ~/.gradle
     ~/.m2
   GLOBAL_CACHE_KEY: gradle-m2-java-11
   GLOBAL_CACHE_RESTORE_KEYS: |
     gradle-m2-java-
     gradle-m2
-  # We cache the source code, resources and build output. This is generated on the build job and reused in dependent jobs.
-  BUILD_CACHE_PATH: ./*
+  # We cache the class files, resources and build output. This is generated on the build job and reused in dependent jobs.
+  BUILD_CACHE_PATH: |
+    .gradle
```

Comment:
Can you verify that this change won't impact workflow re-runs?

The point of all these caches is to have a particular job finish right away if it previously completed successfully and we re-run the whole workflow.

Reply:
I am checking

Reply:
Confirmed